LEAF_Request_Portal/FormWorkflow.php

```diff
@@ -61,177 +63,206 @@ public function isActive(): bool
         return isset($res[0]);
     }
 
-    /**
-     * Retrieves current steps of a form's workflow, controls access to steps
-     * @return array database result
-     * @return null if no database result
-     */
-    public function getCurrentSteps(): array|int|null
-    {
-        // check privileges
-        require_once 'form.php';
-        $form = new Form($this->db, $this->login);
-        if (!$form->hasReadAccess($this->recordID))
-        {
-            return 0;
+    private function getDependency(int $depID) {
+        if(!isset($this->cache["dependencyID{$depID}"])) {
+            $vars = array(':dependencyID' => $depID);
+            $strSQL = 'SELECT * FROM dependency_privs WHERE dependencyID = :dependencyID';
+            $this->cache["dependencyID{$depID}"] = $this->db->prepared_query($strSQL, $vars);
         }
+        
+        return $this->cache["dependencyID{$depID}"];
+    }
 
-        $steps = array();
-        $vars = array(':recordID' => $this->recordID);
-        $strSQL = 'SELECT dependencyID, recordID, stepID, stepTitle, blockingStepID, workflowID, serviceID, filled, stepBgColor, stepFontColor, stepBorder, description, indicatorID_for_assigned_empUID, indicatorID_for_assigned_groupID, jsSrc, userID, requiresDigitalSignature FROM records_workflow_state
+    private function getServiceQuadrads(int $serviceID) {
```

Comment:
Need return type specified.  Also, probably want some documentation on the function.

Reply:
Updated